gammapy/cube/tests/test_edisp_map.py

```diff
@@ -0,0 +1,126 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+import numpy as np
+from numpy.testing import assert_allclose
+import astropy.units as u
+from astropy.units import Unit
+from astropy.coordinates import SkyCoord
+from ...irf import EnergyDispersion2D, EffectiveAreaTable2D
+from ...maps import MapAxis, WcsGeom
+from ...cube import EDispMap, make_edisp_map, make_map_exposure_true_energy
+
+
+def fake_aeff2d(area=1e6 * u.m ** 2):
+    offsets = np.array((0.0, 1.0, 2.0, 3.0)) * u.deg
+    energy = np.logspace(-1, 1, 5) * u.TeV
+    energy_lo = energy[:-1]
+    energy_hi = energy[1:]
+
+    aeff_values = np.ones((4, 3)) * area
+
+    return EffectiveAreaTable2D(
+        energy_lo,
+        energy_hi,
+        offset_lo=offsets[:-1],
+        offset_hi=offsets[1:],
+        data=aeff_values,
+    )
+
+
+def make_edisp_map_test():
+
+    etrue = [0.2, 0.7, 1.5, 2.0, 10.0] * u.TeV
+    migra = np.linspace(0.0, 3.0, 51)
+    offsets = np.array((0.0, 1.0, 2.0, 3.0)) * u.deg
+
+    pointing = SkyCoord(0, 0, unit="deg")
+    energy_axis = MapAxis(
+        nodes=[0.2, 0.7, 1.5, 2.0, 10.0],
+        unit="TeV",
+        name="energy",
+        node_type="edges",
+        interp="log",
+    )
+    migra_axis = MapAxis(nodes=np.linspace(0.0, 3.0, 51), unit="", name="migra")
+
+    edisp2d = EnergyDispersion2D.from_gauss(etrue, migra, 0.0, 0.2, offsets)
+
+    geom = WcsGeom.create(
+        skydir=pointing, binsz=0.2, width=5, axes=[migra_axis, energy_axis]
```

Comment:
Could you use a smaller map for testing? Let's say 5x5 spatial pixels instead of 25 x 25?

Reply:
Done